app/docs/0.11.x/plugin-development/entities-cache.md

```diff
@@ -15,195 +15,275 @@ chapter: 7
-|------------------------------------|---------------------------
-| `ok, err = cache.set(key, value, ttl)`  | Stores a Lua object into the in-memory cache with the specified key (the optional ttl is in seconds). The value can be any Lua type, including tables. Returns `true` or `false`, and `err` if the operation fails.
-| `value = cache.get(key)`           | Retrieves the Lua object stored in a specific key.
-| `cache.delete(key)`                | Deletes the cache object stored at the specified key.
-| `ok, err = cache.sh_add(key, value, ttl) | Adds a new value to the SHM cache, optional ttl in seconds (if `nil` it never expires)
-| `ok, err = cache.sh_set(key, value, ttl) | Sets a new value under the specified key in the SHM, optional ttl in seconds (if `nil` it never expires)
-| `value = cache.sh_get(key)         | returns the value stored in the SHM under the key, or `nil` if not found
-| `cache.sh_delete(key)              | deletes a value from the SHM
-| `newvalue, err = cache.sh_incr(key, amount)` | Increments a number stored in SHM under the specified key, by an amount of units specified. The number needs to be already present in the cache or an error will be returned. If successful, it returns the newly-incremented value, otherwise an error.
-| `value, ... = cache.get_or_set(key, ttl, function, ...)` | This is a utility method that retrieves an object with the specified key, but if the object is `nil` then the passed function will be executed instead, whose return value will be used to store the object at the specified key. This effectively makes sure that the object is only loaded from the datastore one time, since every other invocation will load the object from the in-memory cache.
+Function name                                 | Description
+----------------------------------------------|---------------------------
+`value, err = cache:get(key, opts?, cb, ...)` | Retrieves the value from the closest cache level. If no cache (L1 nor L2) has the value, invokes `cb` in protected mode. `cb` must return one (and only one) value that will be cached. It *can* throw errors, as those will be caught and properly logged by Kong, at the `ngx.ERR` level. This function **does** cache negative results (`nil`). As such, one must rely on its second argument `err` when checking for errors.
+`ttl, err, value = cache:probe(key)`          | Checks if a value is cached in the L2 cache. If it is, returns its remaining TTL. It not, returns `nil`. The value being cached can also be a negative caching. The third return value is the value being cached itself.
+`cache:invalidate_local(key)`                       | Evicts a value from the node's cache (L1 and L2).
+`cache:invalidate(key)`                             | Evicts a value from the node's cache **and** propagates the eviction events to all other nodes in the cluster.
+`cache:purge()`                                     | Evicts **all** values from the node's cache (L1 and L2).
```

Comment:
Yeah that's true - I mostly updated this doc to the new API, mentions of the caching levels exited before, and I'm also biased because of the lua-resty-mlcache required technical documentation 😅